src/figuresClasses.ts

```diff
@@ -2,10 +1,80 @@
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public a: number,
+    public b: number,
+    public c: number,
+    public color: Color,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    const longestSide = Math.max(this.a, this.b, this.c);
```

Comment:
The calculation of 'longestSide' is correct, but ensure that 'this.a', 'this.b', and 'this.c' are used consistently. Currently, 'a', 'b', and 'c' are used in the constructor, which could lead to confusion.